docs/source/caching/http-cache.mdx

```diff
@@ -7,97 +7,78 @@ title: HTTP cache
 
 <Note>
 
-The HTTP cache is only available on Android and the JVM.
+HTTP caching is only available on Android and the JVM.
 
 </Note>
 
+Apollo leverages OkHttp's [cacheUrlOverride](https://square.github.io/okhttp/5.x/okhttp/okhttp3/-request/-builder/cache-url-override.html) to cache GraphQL POST requests when [persisted queries](https://www.apollographql.com/docs/kotlin/advanced/persisted-queries) and GET requests are not an option.
+
 ## Setup
 
-To enable HTTP cache support, add the dependency to your project's `build.gradle` file:
+`OkHttpClient` comes with [built-in cache support](https://square.github.io/okhttp/features/caching/):
 
-```kotlin title="build.gradle[.kts]"
-dependencies {
-  implementation("com.apollographql.apollo:apollo-http-cache:5.0.0-alpha.2")
-}
+```kotlin
+  val okHttpClient = OkHttpClient.Builder()
+      .cache(Cache(
+          directory = File(application.cacheDir, "http_cache"),
+          // $0.05 worth of phone storage in 2020
+          maxSize = 50L * 1024L * 1024L // 50 MiB
+      ))
+      .build()
 ```
 
-If you're targeting Android API level < 26, you'll need to enable [core library desugaring](https://developer.android.com/studio/write/java8-support#library-desugaring) to support the `java.time` API:
+Configure your `ApolloClient` to use your `okHttpClient`:
 
 ```kotlin
-android {
-    compileOptions {
-        // Flag to enable support for the new language APIs
-        coreLibraryDesugaringEnabled = true
-    }
-}
+  val apolloClient = ApolloClient.Builder()
+        .serverUrl(url)
+        .okHttpClient(apolloClient) // highlight-line
+        .addInterceptor(CacheUrlOverrideInterceptor("https://example.com/graphql")) 
```

Comment:
Should this be `CacheUrlOverrideInterceptor(url)`?

```suggestion
        .addInterceptor(CacheUrlOverrideInterceptor(url)) 
```

Actually, if I understand correctly, this could be any URL, it doesn't have to match the server's URL, right? Which could be a bit puzzling :) But I guess it makes more sense if it is the same in the doc.

Reply:
Moving everything to `DefaultHttpRequestComposer` removes this all together since `serverUrl` is also going to be used for cache url override. 
There's always the possibility that our query parameters (`operationName`, `variablesHash`, `operationId`) clash with the user ones. If that happens, they will have to define their own `HttpRequestComposer`